Userland/Libraries/LibGfx/Color.h

```diff
@@ -313,6 +313,20 @@ class Color {
     Vector<Color> shades(u32 steps, float max = 1.f) const;
     Vector<Color> tints(u32 steps, float max = 1.f) const;
 
+    constexpr void saturate_to(double saturation)
+    {
+        auto hsv = to_hsv();
+        auto alpha = this->alpha();
+        *this = Color::from_hsv(hsv.hue, saturation, hsv.value);
+        this->set_alpha(alpha);
+    }
+    constexpr Color saturated_to(double saturation) const
+    {
+        auto color = *this;
+        color.saturate_to(saturation);
+        return color;
+    }
```

Comment:
Color seems to be immutable for the most part, please inline `saturate_to` or make it private

Reply:
There are a bunch of other functions that have this same behavior

Reply:
Namely...? The only self-mutating functions in `Color.h` returning void are `set_<component>`, which is clear from the name.